### PR TITLE
For #42437: Task actions are now cached per entity instead of by type.

### PR DESCRIPTION
**Pull request by thebeeland**
* This allows for different actions to be configured for Tasks linked to Shots vs. those linked to Assets.

* Also adds local file linking file browse/pick RPC methods to api_v2.

### CODE REVIEW

File: python/tk_framework_desktopserver/shotgun/api_v2.py [hunk 1]
```diff
@@ -829,6 +889,29 @@ def _get_subprocess_kwargs(self):
...
+            logger.debug("Parent entity type found in cache for Task %s.", task_id)
+        else:
+            context = sgtk.context.from_entity(
+                self._engine.sgtk,
+                "Task",
+                task_id,
+            )
+            entity_type = context.entity["type"]
+
+            if "task_parent_types" in cache:
```
Comment (jfboismenu) [inline python/tk_framework_desktopserver/shotgun/api_v2.py]: Depending on whether you think the following is cleaner or not, you could also have written this as

```python
cache.setdefault("task_parent_types", {})[task_id] = entity_type
```

Comment (jfboismenu) [inline python/tk_framework_desktopserver/shotgun/api_v2.py]: ```setdefault``` retrieves the value for `task_parent_types`. If it doesn't exist, then it sets `{}` as the value and returns it.

File: python/tk_framework_desktopserver/shotgun/api_v2.py [hunk 2]
```diff
@@ -829,6 +889,29 @@ def _get_subprocess_kwargs(self):
 
         return kwargs
 
+    def _get_task_parent_entity_type(self, task_id):
+        """
+        Gets the Task entity's parent entity type.
+        """
+        cache = self.WSS_KEY_CACHE[self._wss_key]
+
+        if "task_parent_types" in cache and task_id in cache["task_parent_types"]:
```
Comment (jfboismenu) [inline python/tk_framework_desktopserver/shotgun/api_v2.py]: TASK_PARENT_TYPES = "task_parent_types" at the top of the file would be appreciated. This is non standard enough that it is worth constantifying.

File: python/tk_framework_desktopserver/shotgun/api_v2.py [hunk 3]
```diff
@@ -245,10 +255,23 @@ def get_actions(self, data):
                 # It's neither useful to the client, nor json encodable.
                 del pipeline_config["descriptor"]
 
-                lookup_hash = pc_data["lookup_hash"]
+                # In all cases except for Task entities, we'll already have a
+                # lookup hash computed. If we're dealing with a Task, though,
+                # it'll be a None and we'll need to compute is live. This is
```
Comment (jfboismenu) [inline python/tk_framework_desktopserver/shotgun/api_v2.py]: we'll need to compute _IT_ live?

File: python/tk_framework_desktopserver/shotgun/api_v2.py (hunk 2 above)
Comment (thebeeland) [inline python/tk_framework_desktopserver/shotgun/api_v2.py]: Switching all of the cache keys to constants. Good idea.

**Comment by thebeeland** [comment]
Good nitpicks. Taken care of!

**Pull request merged**